node/resolvers/items.ts

```diff
@@ -112,9 +112,15 @@ export const mutations = {
     const cleanItems = items.map(
       ({ options, index, uniqueId, ...rest }) => rest
     )
-    const withOptions = items.filter(
-      ({ options }) => !!options && options.length > 0
-    )
+
+    // Filter only the items with options and ensure that they have a proper index property
+    const withOptions = items.reduce((acc: OrderFormItemInput[], item: OrderFormItemInput, currentIndex: number) => {
+      if (item.options && item.options.length > 0) {
+        return [ ...acc, { ...item, index: item.index ?? currentIndex } ]
+      }
+
+      return acc
+    }, [])
```

Comment:
instead of a reduce, why not just map through the items to add the index before filtering them?
```suggestion
    const withOptions = items
      .map(item => ({ ...item, index: item.index ?? currentIndex }))
      .filter(({ options }) => !!options && options.length > 0)
```

what do you think?

Reply:
Hey @lucasecdb , thanks for the suggestion, but i was trying to avoid an additional loop (`filter` + `map`), doing all in a single roll. But if you guys rather legibility, that's of for me, I just change the order of loops: filter then map, because we map fewer elements.

Reply:
I don't think performance here isn't a concern, even if we have ~300 items in the cart this loop should be rather quick.

About the order of operations, if we invert the filter with map we will have the wrong indexes mapped

Reply:
good point, my bad haha